pkg/skaffold/test/custom/custom.go

```diff
@@ -103,12 +102,12 @@ func runCustomCommand(ctx context.Context, out io.Writer, test latest.CustomTest
 				} else if ctx.Err() == context.Canceled {
 					color.Red.Fprintf(out, "Command cancelled\n")
 				}
-				return ctx.Err()
+				return commandExecutionCancelledOrTimedoutErr(ctx.Err())
```

Comment:
can we split `cmdRunCancelled` and `cmdRunTimeout`?

Reply:
The command exited with either message. The error text needs to be parsed further more to determine whether it is cancelled or timeout.

Reply:
doesn't this work?

```go
if ctx.Err() == context.DeadlineExceeded {
	color.Red.Fprintf(out, "Command timed out\n")
         return cmdRunTimeout(err)
} else if ctx.Err() == context.Canceled {
	color.Red.Fprintf(out, "Command cancelled\n")
	return cmdRunCancelled(err)
}
```

Reply:
Split the execution context into 3 errors:

if ctx.Err() == context.DeadlineExceeded {
	color.Red.Fprintf(out, "Command timed out\n")
        return commandTimedoutErr(ctx.Err())
} else if ctx.Err() == context.Canceled {
	color.Red.Fprintf(out, "Command cancelled\n")
	return commandCancelledErr(ctx.Err())
}
return commandExecutionErr(ctx.Err())